lib/rubystats/binomial_distribution.rb

```diff
@@ -73,76 +73,37 @@ def get_cdf(_x)
       return sum
     end
 
-    # Inverse of the cumulative binomial distribution function 
-    # (equivalent to R qbinom function).
+    # Inverse of the cumulative binomial distribution function     
     # returns the value X for which P(x < _x).
-    def get_icdf(prob)
+    def get_icdf(prob)   
       check_range(prob)
-      (find_root(prob, @n/2, 0.0, @n)).floor
+      sum = 0.0
+      k = 0
+      until prob <= sum 
+        sum += get_pdf(k)
+        k += 1
+      end 
+      return k - 1
```

Comment:
I don't think this `return` is necessary - the last statement is returned implicitly.